### DISCUSSION TITLE
does videojs support full screen mode on iPhone and android devices?

### DISCUSSION THREAD

**Discussion by suryab21** [videojs/video.js#8037] [Q&A]
I am looking for a video plug-in which can play video and YouTube video across multiple platforms (e.g. Mobile, Tablet (iPhone, Android) as well as on Desktop).

I have checked the videojs demo on desktop and it works fine but on mobile I am not sure if the videojs is capable to play videos in full screen mode or not when we click on the full screen button [ ]. 

I know that there are some limitations for full screen mode on iOS safari itself (_https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API#api.document.fullscreen_) but still I am curious to know if somehow with some other way videojs is capable of providing the full screen feature on iOS and android devices for normal as well as YouTube videos. can user enter into full screen mode by clicking the [ ] square button on control bar on iOS/iPhone and Android devices including YouTube videos? What are the videojs limitations here?

@heff, @gkatsev 

Your feedback will be appreciated!

**Comment by gkatsev** [comment]
Fullscreen on Android shouldn't be an issue with youtube or not. Though, for webviews, you have to enable the fullscreen api, I believe, as it isn't available by default.
iPhones do not support the fullscreen api. Regular videos fullscreen on iphone by taking just the media fullscreen. I believe that should work for youtube as well. Video.js also has a "full window" mode, which makes the player the size of the viewport, but it doesn't end up hiding any of phone UI, like the status bar and browser bars.

**Comment by densen2014** [comment]
try this, work fine in my ios

```
    <video id="@Id"
           class="video-js"
           muted
           webkit-playsinline
           playsinline
           x-webkit-airplay="allow"
           x5-video-player-type="h5"
           x5-video-player-fullscreen="true"
           x5-video-orientation="portraint">
        <p class="vjs-no-js">
            To view this video please enable JavaScript, and consider upgrading to a
            web browser that
            <a href="https://videojs.com/html5-video-support/" target="_blank">
                supports HTML5 video
            </a>
        </p>
    </video>
```


my projet:

https://blazor.app1.es/videoPlayers

**Reply by Nick-Raychev** [reply]
Does it work with custom html overlay inside the video.
